Mathlib/Analysis/SpecialFunctions/NthRoot.lean

```diff
@@ -13,10 +13,120 @@ noncomputable section
 
 namespace Real
 
-def nthRoot (n : ℕ) (r : ℝ) : ℝ :=
-  if Even n then r ^ (n⁻¹ : ℝ) else SignType.sign r ^ n * abs r ^ (n⁻¹ : ℝ)
+section qpow
 
-theorem nthRoot_of_even {n : ℕ} (hn : Even n) (r : ℝ) : nthRoot n r = r ^ (n⁻¹ : ℝ) :=
+instance instQPow : Pow ℝ ℚ where
+  pow r q :=
+    if Even q.den then r ^ (q : ℝ) else SignType.sign r ^ (q.den * q.num) * abs r ^ (q : ℝ)
+
+theorem qpow_of_even (r : ℝ) {q : ℚ} (hn : Even q.den) : r ^ q = r ^ (q : ℝ) :=
```

Comment:
rename: `qpow_eq_rpow_of_even` or `qpow_eq_of_even`. Also in many lemmas below. If a hypothesis is mentioned, the conclusion should be described using `qpow_eq` or something more specific than that.